### ISSUE TITLE
Feature Request: `function-component-definition` prefer function declaration for default exports

### ISSUE THREAD

**Issue by Avasam** [jsx-eslint/eslint-plugin-react#3969]
Maybe there's a good reason that I don't see, but having to use an arrow function for a default export component just leads to extra boilerplate and has been an annoyance for me:

```jsx
const MyComponent = () => {
  // ...
}
// These can be hundreds of lines appart
export default MyComponent
```
vs
```jsx
export default function MyComponent() {
  // ...
}
```

but I don't want simple braceless arrow function named components to suddenly have to require braces and `return`:

```jsx
function MyInnerComponent() {
  return <></>
}

export function MyComponent() {
  return <MyInnerComponent />
}
```
vs
```jsx
const MyInnerComponent= () => <></>
export const MyComponent= () => <MyInnerComponent />
```

**Comment by ljharb** [comment]
You definitely shouldn't ever use an arrow function for a component - which is what the https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md helps enforce.

**Comment by Avasam** [comment]
> You definitely shouldn't ever use an arrow function for a component

"definitely shouldn't ever" are strong words, but other than claiming consistency, that doc does not explain why. Leaving me still wondering.

**Comment by ljharb** [comment]
Primarily because arrow functions were designed to be a replacement for inline callbacks. Separate things should be actual named functions (inferred names don't count). There's not really a technical reason - just a philosophical one.
